File: Hard/34. Find First and Last Position of Element in Sorted Array/sol.py
class Solution:
    def searchRange(self, N, T):
        Tleft = self.find(T, N, 0)
        if Tleft == len(N) or N[Tleft] != T:
            return [-1, -1]
        return [Tleft, self.find(T + 1, N, Tleft) - 1]

    def find(self, target, arr, left):
        right = len(arr) - 1
        while left <= right:
            mid = (left + right) // 2
            if arr[mid] < target:
                left = mid + 1
            else:
                right = mid - 1
        return left

File: Hard/34. Find First and Last Position of Element in Sorted Array/sol.java
import java.util.*;

public class sol {
    public static void main(String[] args) {
        // call the fn here
    }

    class Solution {
        public int[] searchRange(int[] N, int T) {
            int Tleft = find(T, N, 0);
            if (Tleft == N.length || N[Tleft] != T) return new int[] {
                -1, -1
            };
            return new int[] {
                Tleft,
                find(T + 1, N, Tleft) - 1
            };
        }
        public int find(int target, int[] arr, int left) {
            int right = arr.length - 1;
            while (left <= right) {
                int mid = left + right >> 1;
                if (arr[mid] < target) left = mid + 1;
                else right = mid - 1;
            }
            return left;
        }
    }
}
